src/librustc_back/target/armv7_unknown_linux_gnueabihf.rs

```diff
@@ -13,7 +13,7 @@ use target::{Target, TargetOptions, TargetResult};
 pub fn target() -> TargetResult {
     let base = super::linux_base::opts();
     Ok(Target {
-        llvm_target: "armv7-unknown-linux-gnueabihf".to_string(),
+        llvm_target: "arm-unknown-linux-gnueabihf".to_string(),
```

Comment:
Not sure if it matters, but you lose the `Tag_CPU_arch_profile: Application` attribute this way, but you could add it back with `+aclass`.
IMO `armv7-unknown-linux-gnueabihf` with `-neon,+vfp3,+d16` is all that is needed here.